.github/workflows/cd-pipeline.yml

```diff
@@ -192,17 +193,95 @@ jobs:
             git push --tags
           fi
+  push-promotion:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs:
+      - get-chart
+      - update-index
+    name: Push Promotion files
+    if: ${{ needs.get-chart.outputs.result == 'ok' }}
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29
+        with:
+          path: charts
+      - name: Install helm
+        run: |
+          HELM_TARBALL="helm-v3.8.1-linux-amd64.tar.gz"
+          curl -SsLfO "https://get.helm.sh/${HELM_TARBALL}" && sudo tar xf "$HELM_TARBALL" --strip-components 1 -C /usr/local/bin
+      - env:
+          CHART_NAME: ${{ needs.get-chart.outputs.chart }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_PROMOTION_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_PROMOTION_SECRET_ACCESS_KEY }}
+          AWS_PROMOTION_BUCKET: ${{ secrets.AWS_PROMOTION_BUCKET }}
+          AWS_MAX_ATTEMPTS: 3
+          AWS_DEFAULT_REGION: us-east-1
+        run: |
+          #!/bin/bash
+          chart="bitnami/${CHART_NAME}"
+          cd charts
+          # Get chart and app version
+          chart_version="$(yq e '.version' "${chart}/Chart.yaml")"
+          app_version="$(yq e '.appVersion' "${chart}/Chart.yaml")"
+          # Get image list
+          helm dep build ${chart}
+          image_list="$(yq '.annotations.images' "${chart}/Chart.yaml" | yq '[ .[] | .image ] | tojson')"
+          for dependency in "${chart}/charts/"*.tgz; do
+            chart_yaml_file="$(tar -tzf "$dependency" | grep -E "^[^/]+/Chart.yaml$")"
+            if [[ -n "${chart_yaml_file}" ]]; then
+              dependency_image_list="$(tar -xzO -f "$dependency" "$chart_yaml_file" | yq '.annotations.images' | yq '[ .[] | .image ] | tojson')"
+              image_list="$(jq -c --null-input --argjson arr1 "$image_list" --argjson arr2 "$dependency_image_list" '$arr1 + $arr2 | unique')"
+            fi
+          done
+          # Build JSON files
+          release_date="$(date -u +"%Y/%m/%d")"
+          file_prefix="$(date -u +"%s%3N-${CHART_NAME}-${app_version}")"
+          json_template=$(cat << "EOF"
+          {
+            "platform_id": $platform_id,
+            "application": $app,
+            "external_id": "\($app):\($chart_version)",
+            "version": $app_version,
+            "bundled_os_version": $bundled_os_version,
+            "properties": {
+                "chart_url": "https://charts.bitnami.com/bitnami/\($app)-\($chart_version).tgz",
+                "containers": $image_list,
+                "github_repository": "bitnami/charts/tree/main/bitnami/\($app)",
+            }
+          }
+          EOF
+          )
+          jq --null-input \
+            --arg platform_id "bitnami-chart-debian-x64" \
+            --arg app "${CHART_NAME}" \
+            --arg app_version "${app_version}" \
+            --arg chart_version "${chart_version}" \
+            --arg bundled_os_version "12" \
```

Comment:
Instead of hardcoding this arg, do you think it would be possible to parameterise it?

Then I don't know if it should be a matter of automatically detecting it or simply trigger an error if version is not 12/bookworm.

I'm worried we may forget about this in the future and continue submitting the charts JSON files with incorrect information

Reply:
I couldn't agree more! Thanks for the heads up, I completely forgot it.

Reply:
We need to merge it, I will continue with this in another PR